sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_connection_async.py

```diff
@@ -129,53 +151,40 @@ def __init__(self, endpoint, **kwargs):  # pylint:disable=too-many-statements
-        self._idle_timeout = kwargs.pop("idle_timeout", None)  # type: Optional[int]
-        self._outgoing_locales = kwargs.pop(
-            "outgoing_locales", None
-        )  # type: Optional[List[str]]
-        self._incoming_locales = kwargs.pop(
-            "incoming_locales", None
-        )  # type: Optional[List[str]]
-        self._offered_capabilities = None  # type: Optional[str]
-        self._desired_capabilities = kwargs.pop(
-            "desired_capabilities", None
-        )  # type: Optional[str]
-        self._properties = kwargs.pop(
-            "properties", None
-        )  # type: Optional[Dict[str, str]]
+        self._max_frame_size: int = max_frame_size
+
+        self._remote_max_frame_size: Optional[int] = None
+        self._channel_max: int = channel_max
+        self._idle_timeout: Optional[float] = idle_timeout
+        self._outgoing_locales: Optional[List[str]] = outgoing_locales
+        self._incoming_locales: Optional[List[str]] = incoming_locales
+        self._offered_capabilities: Optional[List[str]] = offered_capabilities
+        self._desired_capabilities: Optional[List[str]] = desired_capabilities
+        self._properties: Optional[Dict[str, Any]] = properties
 
         self._remote_properties: Optional[Dict[str, str]] = None
 
-        self._allow_pipelined_open = kwargs.pop(
-            "allow_pipelined_open", True
-        )  # type: bool
-        self._remote_idle_timeout = None  # type: Optional[int]
-        self._remote_idle_timeout_send_frame = None  # type: Optional[int]
-        self._idle_timeout_empty_frame_send_ratio = kwargs.get(
-            "idle_timeout_empty_frame_send_ratio", 0.5
+        self._allow_pipelined_open: bool = allow_pipelined_open
+        self._remote_idle_timeout: Optional[float] = None
+        self._remote_idle_timeout_send_frame: Optional[float] = None
+        self._idle_timeout_empty_frame_send_ratio: float = (
+            idle_timeout_empty_frame_send_ratio
         )
-        self._last_frame_received_time = None  # type: Optional[float]
-        self._last_frame_sent_time = None  # type: Optional[float]
-        self._idle_wait_time = kwargs.get("idle_wait_time", 0.1)  # type: float
-        self._error = None
-        self._outgoing_endpoints = {}  # type: Dict[int, Session]
-        self._incoming_endpoints = {}  # type: Dict[int, Session]
-
-    async def __aenter__(self):
+        self._last_frame_received_time: Optional[float] = None
+        self._last_frame_sent_time: Optional[float] = None
+        self._idle_wait_time: float = idle_wait_time
+        self._error: Optional[AMQPConnectionError] = None
+        self._outgoing_endpoints: Dict[int, Session] = {}
+        self._incoming_endpoints: Dict[int, Session] = {}
+
+    async def __aenter__(self) -> "Connection":
         await self.open()
         return self
 
-    async def __aexit__(self, *args):
+    async def __aexit__(self, *args) -> None:
```

Comment:
```suggestion
    async def __aexit__(self, *args: Any) -> None:
```

Reply:
why are there args when we don't pass them anywhere?

Reply:
It's part of the protocol for a context manager.
And these args will be populated in the case that the context exited with an exception. It's just that our SDKs don't use them.